src/modules/menu/components/LayerCatalogue.vue

```diff
@@ -13,22 +14,73 @@ const { layerCatalogue, compact } = defineProps({
         default: false,
     },
 })
+const { layerCatalogue, compact } = toRefs(props)
 
 const store = useStore()
 
+const searchText = ref('')
+const searchInput = ref(null)
+
+const showSearchBar = computed(() => layerCatalogue.value.length > 0)
+
 function clearPreviewLayer() {
     if (store.state.layers.previewLayer) {
         store.dispatch('clearPreviewLayer')
     }
 }
+
+let debounceTimeout = null
+function onSearchInput(event) {
+    clearTimeout(debounceTimeout)
+    if (event.target.value?.length >= 2) {
+        debounceTimeout = setTimeout(() => {
+            searchText.value = event.target.value
+        }, 100)
+    } else if (!event.target.value) {
+        searchText.value = ''
+    }
+}
+
+function clearSearchText() {
+    searchText.value = ''
+    searchInput.value.focus()
+}
 </script>
 
 <template>
-    <div class="menu-catalogue-list" @mouseleave="clearPreviewLayer">
+    <div @mouseleave="clearPreviewLayer">
+        <div v-if="showSearchBar" class="input-group input-group-sm d-flex p-1">
+            <span id="searchCatalogueIcon" class="input-group-text">
+                <FontAwesomeIcon :icon="['fas', 'search']" />
+            </span>
+            <input
+                ref="searchInput"
+                type="text"
+                class="form-control"
+                :class="{ 'rounded-end': !searchText.length }"
+                :placeholder="$t('search_in_catalogue_placeholder')"
```

Comment:
please replace `$t` by `i18n.t` and importing it as `const i18n = useI18n()` above 😉

Reply:
My bad copy paste habits 😉